sql-to-dbsp-compiler/lib/sqllib/src/lib.rs

```diff
@@ -968,6 +993,95 @@ pub fn cbrt_d(value: F64) -> F64 {
 
 some_polymorphic_function1!(cbrt, d, F64, F64);
 
+/////////// Hyperbolic Fucntions //////////////
+
+#[inline(always)]
+pub fn sinh_d(value: F64) -> F64 {
+    value.into_inner().sinh().into()
+}
+
+some_polymorphic_function1!(sinh, d, F64, F64);
+
+#[inline(always)]
+pub fn cosh_d(value: F64) -> F64 {
+    value.into_inner().cosh().into()
+}
+
+some_polymorphic_function1!(cosh, d, F64, F64);
+
+#[inline(always)]
+pub fn tanh_d(value: F64) -> F64 {
+    value.into_inner().tanh().into()
+}
+
+some_polymorphic_function1!(tanh, d, F64, F64);
+
+#[inline(always)]
+pub fn coth_d(value: F64) -> F64 {
+    let tanh = value.into_inner().tanh();
+    if tanh.is_zero() {
+        f64::INFINITY.into()
+    } else {
+        (1.0 / tanh).into()
+    }
+}
+
+some_polymorphic_function1!(coth, d, F64, F64);
+
+#[inline(always)]
+pub fn asinh_d(value: F64) -> F64 {
+    value.into_inner().asinh().into()
+}
+
+some_polymorphic_function1!(asinh, d, F64, F64);
+
+#[inline(always)]
+pub fn acosh_d(value: F64) -> F64 {
+    if value.into_inner() < 1.0 {
+        panic!("input out of range")
+    }
+
+    value.into_inner().acosh().into()
+}
+
+some_polymorphic_function1!(acosh, d, F64, F64);
+
+#[inline(always)]
+pub fn atanh_d(value: F64) -> F64 {
+    let inner = value.into_inner();
+    if !(-1.0..=1.0).contains(&inner) && !inner.is_nan() {
+        panic!("input out of range")
```

Comment:
the error message should probably be more informative, including the actual value and the range.
we'll also add position at some point.

Reply:
Ah okay, will add the value to the error message.